src/messaging/utils.ts

```diff
@@ -21,11 +21,11 @@ import { UUID } from '@kiltprotocol/utils'
  * @returns The message created.
  */
 
-export function fromBody(
-  body: MessageBody,
+export function fromBody<Body extends MessageBody>(
```

Comment:
MessageBody is the union of credential api message types though, right? You'd need `IMessageBodyBase` then. Honestly we may want to improve the naming if we're confusing ourselves now. How about `MessageBody` for the generic type (formerly known as `IMessageBodyBase`) and `CredentialApiMessageBody` for the union of credential api message body types?

Reply:
updated types and function definition in: [a93d110](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/a93d11054e6275f556670f92f40169417e71550b) 

Is KiltApiMessageBody not a better name?

Reply:
The function uses `MessageBody` which is the renamed definition of `MessageBodyBase`